### PR TITLE
Servicetype to metadata

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java [hunk 1]
```diff
@@ -9,9 +9,10 @@
 import org.myrobotlab.framework.MRLListener;
 import org.myrobotlab.framework.MethodEntry;
 import org.myrobotlab.framework.Outbox;
+import org.myrobotlab.service.interfaces.ServiceLifeCycleListener;
 
 public interface ServiceInterface
-    extends ServiceQueue, LoggingSink, NameTypeProvider, MessageSubscriber, MessageSender, StateSaver, Invoker, StatePublisher, StatusPublisher, ServiceStatus, Attachable {
+    extends ServiceLifeCycleListener, ServiceQueue, LoggingSink, NameTypeProvider, MessageSubscriber, MessageSender, StateSaver, Invoker, StatePublisher, StatusPublisher, ServiceStatus, Attachable {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java:15]: seems like the service interface should implement the ServiceLifeCycle interface... unless you're saying generically that a service can generically listen to another services lifecycle..  is that what you're intending here?

File: src/main/java/org/myrobotlab/service/Runtime.java [hunk 2]
```diff
@@ -105,7 +106,7 @@
  * check for 64 bit OS and 32 bit JVM is is64bit()
  *
  */
-public class Runtime extends Service implements MessageListener, RemoteMessageHandler, Gateway, LocaleProvider {
+public class Runtime extends Service implements MessageListener, ServiceLifeCyclePublisher, RemoteMessageHandler, Gateway, LocaleProvider {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Runtime.java:109]: why is runtime the only thing that is publishing lifecycle stuff?  i mean. i'd expect services to implement a SerivceLifeCycle  interface (start/stop/created..etc..)    .. i guess services could publish when their lifecycle was called to notify other listeners, however that should happen after the service processes it's lifecycle request.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractMotor.java [hunk 3]
```diff
@@ -26,9 +26,7 @@
 
 package org.myrobotlab.service.abstracts;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractMotor.java:27]: changes to motor control should be done in a different pull request. please.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java [hunk 4]
```diff
@@ -24,6 +24,7 @@
 import org.myrobotlab.service.interfaces.KeyConsumer;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java:24]: Changes to speech synthesis should be done in a different pull request from the service life cycle. please

File: src/main/java/org/myrobotlab/service/interfaces/JoystickListener.java [hunk 5]
```diff
@@ -1,10 +1,9 @@
 package org.myrobotlab.service.interfaces;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/JoystickListener.java:1]: changes to joystick interface should be done in a different pull request from service lifecycle please.

File: src/main/java/org/myrobotlab/service/interfaces/ServiceLifeCyclePublisher.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package org.myrobotlab.service.interfaces;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/ServiceLifeCyclePublisher.java:1]: This should probably be renamed to ServiceLifeCycle not publisher. the publisher interface would be "publishStarted"   not  "started"

File: src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java:15]: correct - the abstract base class Service now implements a set of callbacks,
the "interested" services can simply override these if they want to handle the notification events of
registered, created, started, stopped and released.
If they are not interested, they simply do not override.

File: src/main/java/org/myrobotlab/service/Runtime.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/Runtime.java:109]: Some events can be handled inside a class - started and stopped
Other events cannot be handled inside any service .. e.g. registered, released, created
The one thing other services are guaranteed is runtime exists - therefore since registered, released, and created are already handled by runtime, it was logical that it be responsible for broadcasting the remainder of the events started and stopped

File: src/main/java/org/myrobotlab/service/interfaces/ServiceLifeCyclePublisher.java (hunk 6 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/ServiceLifeCyclePublisher.java:1]: Sure .. I'll change it to ServiceLifeCycle.  Its a publishing interface, however registered and released have been around for bajillion years (as string values), so even though it is a publishing interface I'll wait to refactor to publishRegistered etc for another refactor)

File: src/main/java/org/myrobotlab/service/abstracts/AbstractMotor.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractMotor.java:27]: These file only get changed updated, because I'm keeping it in sync with develop.  These are merge conflicts that have been resolved, because changes occurred on this file and were pushed into develop from another pr.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java (hunk 4 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java:24]: These file only get changed updated, because I'm keeping it in sync with develop.  These are merge conflicts that have been resolved, because changes occurred on this file and were pushed into develop from another pr.

File: src/main/java/org/myrobotlab/service/interfaces/JoystickListener.java (hunk 5 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/JoystickListener.java:1]: These file only get changed updated, because I'm keeping it in sync with develop.  These are merge conflicts that have been resolved, because changes occurred on this file and were pushed into develop from another pr.

**Comment by kwatters** [comment]
Sorry pull request too large.  If you want to do MetaData / ServiceType change, do that as a separate pull request.
